main.go

```diff
@@ -155,13 +158,18 @@ func main() {
 		router.Handle("/", createReverseProxy("homepage", homepageControllerURL))
 	}
 
+	if cfg.EnableProfiler {
+		profilerChain := alice.New(profiler.Middleware(cfg.PprofToken)).Then(http.DefaultServeMux)
```

Comment:
So I fully understand this - this middleware is only applicable to this endpoint?

Reply:
The pprof package import `_ "net/http/pprof"` does some magic and wires up the `/debug/...` handlers. These can't be hit when toggled off, due to all routes being directed to reverse proxies. Only when we toggle on the profiler and catch the `/debug` requests do we pass it through the middleware and onto the default mux handler (where the pprof handler are wired in)